### PR TITLE
BIGCZ Map/Sidebar interaction improvements

### PR DESCRIPTION
**Pull request by caseycesari**
## Overview

Several improvements were made to the interaction between the map and
the results sidebar. They include:

- Hovering over features on the map now highlight the feature on the map
  in the same way that hovering over the result in the list highlights the
  corresponding map feature.
- Hovering over the feature on the map will also now highlight the
  corresponding result in the list.
- For WDC, if a popup is opened for a marker, the marker will remain
  highlighted until that popup is closed. Hovering over other features
  on the map will not remove the highlight of the marker with the popup.

Connects #2181

### Demo

bigcz4

### Notes

This PR does not include the second bullet point from #2181 for CINERGI, nor the last bullet point, which is only relevant to CINERGI.

## Testing Instructions

- Make a search and load the WDC tab. 
- Test each of the features listed above and verify they work as described.

### CODE REVIEW

File: src/mmw/js/src/core/views.js [hunk 1]
```diff
@@ -712,8 +712,9 @@ var MapView = Marionette.ItemView.extend({
         });
     },
 
-    createDataCatalogShape: function(geom) {
-        var geomId = geom && geom.properties.id,
+    createDataCatalogShape: function(result) {
+        var geom = result.get('geom'),
+            geomId = geom && geom.properties.id,
```
Comment (arottersman) [inline src/mmw/js/src/core/views.js]: This id is the same as `result.get('id')`. Now that we're passing the whole result around, consider removing the geojson augmentation that happens [here.](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/data_catalog/models.js#L243)

File: src/mmw/js/src/core/views.js [hunk 2]
```diff
@@ -727,64 +728,88 @@ var MapView = Marionette.ItemView.extend({
...
+                layer.on('mouseover', function() {
+                    layer.setStyle(dataCatalogActiveStyle);
+                    result.set('active', true);
+                });
+
+                layer.on('mouseout', function() {
+                    if (geom.type === 'Point') {
+                        // Preserve highlight of marker if popup is open.
+                        // It will get restyled when the popup is closed.
+                        if (!layer._popup._isOpen) {
```
Comment (arottersman) [inline src/mmw/js/src/core/views.js]: The feature works but there's a console error when you hover over the point in the details view now:
screen shot 2017-09-07 at 11 34 05 am

**Comment by arottersman** [comment]
also there are merge conflicts

File: src/mmw/js/src/core/views.js (hunk 2 above)
Comment (caseycesari) [reply src/mmw/js/src/core/views.js]: Ah good catch. I'll look into this.

**Comment by rajadain** [comment]
I can take a look as soon as the merge conflicts are resolved.

**Comment by caseycesari** [comment]
> Also, curious as to why it was hard to do the new popover highlight treatment on CINERGI...will it not be the same as what's done for WDC?

Unfortunately, it doesn't appear so. The public and private properties are different for the WDC and CINERGI layers. I believe this is because one is constructed as GeoJSON, while the other is constructed from CircleMarkers. In Leaflet 0.7, the only way to detect if a popup is open for a particular layer is to access some of the feature's private properties ( This was resolved for v1.0 https://github.com/Leaflet/Leaflet/issues/2494). However, the WDC implementation doesn't work for CINERGI because the `layer._popup` property isn't consistently available for the CINERGI layer, like it is for the WDC layer, even when the popup is open.

Also, the fourth bullet point from #2181 is blocked until the above is implemented.

**Comment by caseycesari** [comment]
Conflicts fixed. Working on the suggested changes.

File: src/mmw/js/src/core/views.js (hunk 1 above)
Comment (caseycesari) [reply src/mmw/js/src/core/views.js]: Addressed in e2fb98d.

File: src/mmw/js/src/core/views.js (hunk 2 above)
Comment (caseycesari) [reply src/mmw/js/src/core/views.js]: Addressed in e2fb98d.

**Comment by caseycesari** [comment]
Ready for another look.

**Comment by rajadain** [comment]
Taking a look now.

**Comment by rajadain** [comment]
The second bullet-point, of the sidebar item staying highlighted when the popup is open, doesn't seem to be working for WDC either:

**Comment by rajadain** [comment]
If the 2 bullet points depend on #2214, I'm open to merging this with what we have and making a follow up card with those two points.

**Comment by caseycesari** [comment]
> The second bullet-point, of the sidebar item staying highlighted when the popup is open, doesn't seem to be working for WDC either:

Fixed and pushed. Thanks for the catch. Accidentally rebased this specific change away earlier.

**Comment by caseycesari** [comment]
> We should make a follow up card for the other work post Leaflet upgrade.

Will do.

**Comment by caseycesari** [comment]
Thanks.

**Pull request merged**